src/subscription/SubscriptionPage.jsx

```diff
@@ -69,49 +77,27 @@ export class SubscriptionPage extends React.Component {
           />
         </h1>
         <div className="col-md-5 pr-lg-5 col-basket-summary">
-          <SubscriptionCart />
+          <SubscriptionDetails />
         </div>
         <div className="col-md-7 pl-lg-5 checkout-wrapper">
-          <Checkout isSubscription />
+          <SubscriptionCheckout />
         </div>
       </div>
     );
-  }
-
-  render() {
-    return (
-      <div className="page__payment container-fluid py-5">
-        <FormattedAlertList
-          summaryQuantity={this.props.summaryQuantity}
-          summarySubtotal={this.props.summarySubtotal}
-        />
-        {this.renderContent()}
-      </div>
-    );
-  }
-}
-
-SubscriptionPage.contextType = AppContext;
-
-SubscriptionPage.propTypes = {
-  intl: intlShape.isRequired,
-  isEmpty: PropTypes.bool,
-  isRedirect: PropTypes.bool,
-  fetchBasket: PropTypes.func.isRequired,
-  summaryQuantity: PropTypes.number,
-  summarySubtotal: PropTypes.number,
-};
-
-SubscriptionPage.defaultProps = {
-  isEmpty: false,
-  isRedirect: false,
-  summaryQuantity: undefined,
-  summarySubtotal: undefined,
+  };
+
+  return (
+    <div className="subscription-page page__payment container-fluid py-5">
+      <FormattedAlertList
+        summaryQuantity={summaryQuantity}
+        summarySubtotal={summarySubtotal}
+      />
+      {renderContent()}
+      <ConfirmationModal />
```

Comment:
Should this component only render on a specific condition? (Also added a comment at the component level)

Reply:
passed `isVisible` prop from parent that can handle visibility in case of successful payment flow later.